firmware/shared/util/os.h

```diff
@@ -0,0 +1,21 @@
+/// @author Blake Freer
+/// @date 2023-11-08
+
+#pragma once
+
+#include "shared/util/noncopyable.h"
+
+namespace shared::util {
+
+// Empty for now
+class os {
+
+};
+
+// TODO: Add more verbose error cases
+enum class OsStatus : int {
+    kOsOk = 0,
```

Comment:
don't need to repeat "Os" in the enum variable names since it is already mentioned in the enum class name
i.e. `OsStatus::kOsOk` is redundant, can use `OsStatus::kOk`